rd-kt/rd-core/src/main/kotlin/com/jetbrains/rd/util/string/RName.kt

```diff
@@ -16,6 +16,21 @@ class RName private constructor(val parent:RName?, val localName: String, val se
      */
     fun sub(localName: String, separator: String) = RName(this, localName, separator)
 
+    fun getNonEmptyRoot(): RName {
```

Comment:
An interesting consequence of this implementation is that `Empty.getNonEmptyRoot()` would be equal to `Empty`. I don't know how to solve this (or whether we even want to "solve" it), though.

Reply:
Ideally, I would prefer to reimplement RName with a reversed list because the only case when straight order is needed is logging. But I don't want to change RName dramatically here, so I ended up with linear complexity implementations for every operation. It shouldn't be a performance problem since we don't have too long chains in RName.